.github/workflows/checks.yaml

```diff
@@ -190,4 +190,5 @@ jobs:
         run: |
           . venv/bin/activate
           cd doc
-          make html
+          pre-commit run --hook-stage push sphinx-generated-doc --all-files || { echo "Make sure that there \
```

Comment:
How does this work? This won't exit if there are any added files right? We need to do some kind of `git status` for that?

Reply:
It's launched all the time, you can try ``pre-commit run --all-fies`` locally, there's just some additional option for manual hooks.

Reply:
No, I meant: how does this fail if the doc hasn't been updated properly? Won't this only fail of the `make` commands throws an error?

Reply:
Pre-commit fail is a file was modified during the run (it's dealing with autoformatter often).